mantidimaging/gui/widgets/spectrum_widgets/roi_form_widget.py

```diff
@@ -37,3 +42,164 @@ def set_binning_visibility(self) -> None:
         self.bin_size_spinBox.setHidden(hide_binning)
         self.bin_step_label.setHidden(hide_binning)
         self.bin_step_spinBox.setHidden(hide_binning)
+
+
+class ROIPropertiesTableWidget(BaseWidget):
+    spin_left: QSpinBox
+    spin_right: QSpinBox
+    spin_top: QSpinBox
+    spin_bottom: QSpinBox
+    label_height: QLabel
+    label_width: QLabel
+    group_box: QGroupBox
+
+    roi_changed = pyqtSignal()
+
+    def __init__(self, parent=None):
+        super().__init__(parent, ui_file='gui/ui/roi_properties_table_widget.ui')
+
+        self.spin_boxes = [self.spin_left, self.spin_right, self.spin_top, self.spin_bottom]
+
+        for spin_box in self.spin_boxes:
+            spin_box.valueChanged.connect(self.roi_changed.emit)
+
+    def set_roi_name(self, name: str) -> None:
+        self.group_box.setTitle(f"Roi Properties: {name}")
+
+    def set_roi_values(self, roi: SensibleROI) -> None:
+        with QSignalBlocker(self):
+            self.spin_left.setValue(roi.left)
+            self.spin_right.setValue(roi.right)
+            self.spin_top.setValue(roi.top)
+            self.spin_bottom.setValue(roi.bottom)
+            self.label_width.setText(str(roi.width))
+            self.label_height.setText(str(roi.height))
+
+    def set_roi_limits(self, shape: tuple[int, ...]) -> None:
+        self.spin_left.setMaximum(shape[1])
+        self.spin_right.setMaximum(shape[1])
+        self.spin_top.setMaximum(shape[0])
+        self.spin_bottom.setMaximum(shape[0])
+
+    def as_roi(self) -> SensibleROI:
```

Comment:
Can we give this a more descriptive name. At a glance elsewhere in the code, I would struggle to interpret the what `as_roi`'s purpose may be

Reply:
I've been doing too much rust. Would `to_roi` be better? Like `to_string` in c++.

Reply:
ahh, I see! yes, to would make feel more intuitive